xla/service/cpu/ir_emitter.cc

```diff
@@ -115,7 +116,13 @@ class IrEmitter::CpuElementalIrEmitter : public ElementalIrEmitter {
                         IrEmitter* ir_emitter, llvm::Module* module)
       : ElementalIrEmitter(
             module, ir_emitter->b(),
-            Options{/*xla_cpu_use_truncate_f32_to_bf16_conversion=*/true}),
+            Options { /*xla_cpu_use_truncate_f32_to_bf16_conversion=*/
+#if defined(INTEL_MKL) && defined(ENABLE_ONEDNN_V3)
+                      !IsNativeConvertSupportedOnThisCPU()
+#else
+                      true
+#endif
```

Comment:
I think we should do this in the current CL. Even as is `IsNativeConvertSupported` seems pretty generic. It may need to be extend in the future, but that's OK.

Let's please move `IsNativeConvertSupported` directly in `cpu/ir_emitter.cc` and call it unconditionally.

Reply:
@kanvi-nervana Could you please implement the change above (or comment if you think that's not a good idea).

Another issue: `IsNativeConvertSupported` tests the CPU we're compiling on, not necessarily the target CPU. Could the two be different?

Reply:
> Another issue: IsNativeConvertSupported tests the CPU we're compiling on, not necessarily the target CPU. Could the two be different?

Yes, they could for AOT mode (although the current thunk runtime doesn't support this mode yet). 

@kanvi-nervana Please check the features string from IrEmitter's [target_machine_features()](https://github.com/openxla/xla/blob/acc5b9d4261328b986e5aedc83301cf21aa55abf/xla/service/cpu/ir_emitter.h#L205) instead. You can get the string by calling [getTargetFeatureString()](https://github.com/llvm/llvm-project/blob/556ec4a7261447d13703816cd3730a891441e52c/llvm/include/llvm/Target/TargetMachine.h#L128) from [target_machine_](https://github.com/openxla/xla/blob/263f197f5c2e97a884907abc6cf7b26b5a6c01b5/xla/service/cpu/target_machine_features.h#L118).

Since we are adding a new function to check from feature string, it's probably good to add a small unit test too for the function as well (feeding in feature strings with and without "amx-bf16" and "avxneconvert").

Reply:
Thanks for the review, I will update the PR with the requested changes.